### DISCUSSION TITLE
NetPad v0.8.0

### DISCUSSION THREAD

**Discussion by tareqimbasher** [tareqimbasher/NetPad#266] [Announcements]
**If you like NetPad, please star the repo :star2:!**

### New Database Connection Providers

You can now add MySQL and MariaDB connections in NetPad. Special thanks to @SimonNyvall!

### A Native Shell (vNext Preview)

This update introduces a new native shell built using Tauri, a Rust-based framework, that greatly reduces memory usage, startup time, and app size!

The new shell substitutes the Electron.js shell NetPad has been using so far. You will find new installers available that use the new native shell with names starting with `netpad_vnext`. Electron shell installers are still available, and will continue being distributed until things are stable with vNext. Both versions of NetPad can be installed and run side-by-side. It would be really helpful if you could please provide your feedback in an issue (if you're having trouble), in [this discussion](https://github.com/tareqimbasher/NetPad/discussions/266) or on [Discord](https://discord.gg/FrgzNBYQFW).

If you'd like to read more about this change, take a look at [this issue](https://github.com/tareqimbasher/NetPad/issues/237). Here are some stats:

#### App Size

| Metric  | Electron (MB) | Tauri (MB) | % Reduction |
| ------------- | ------------- | ------------- | ------------- |
| Shell Size  | 251  | 13.6 | 94.58 |
| .NET App Size  | 192 | 103 | 46.35 |
| Installer Size (.deb)  | 125 | 43 | 65.60 |
| Installed Size (.deb)  | 443 | 111 | 74.94 |

#### Memory Usage

| Metric  | Electron (MB) | Tauri (MB) | % Reduction |
| ------------- | ------------- | ------------- | ------------- |
| Shell  | 230  |34 | 85.22 |
| .NET App  | 110 | 110 | -- |
| OmniSharp  | 104 | 104 | -- |
| Total  | 444 | 248 | 44.14 |

#### Known Limitations with vNext

- If you have your NetPad settings configured to use the native titlebar instead of the integrated one, menu auto-hide does not work yet (not an issue on macOS).
- The main window does not currently remember its previous position and size when you close and restart NetPad.
- The Electron version of NetPad will show you a desktop notification when a script finishes execution and NetPad is in the background. vNext currently does not.
- Some Linux installers are not yet available for vNext, namely: `pacman`, `flatpak` and `snap`
- If you're on Linux and when you launch the app you get an empty window, try disabling DMABUF: `export WEBKIT_DISABLE_DMABUF_RENDERER=1`. This is an upstream bug with NVIDIA proprietary drivers on some installations. If that doesn't work, try one of these instead:
  - `WEBKIT_DISABLE_COMPOSITING_MODE=1`
  - `__NV_PRIME_RENDER_OFFLOAD=1`

Fixes for some or all of the above will be included in the next release.

### Others

- Removed .NET 6 dependency: having a .NET 6 runtime or SDK installed is no longer required.
- If you want to use .NET 9 in your scripts you will need .NET 9 Preview 3 SDK or later installed.
- Improvements to data connection scaffolding.
- New scripts will use the latest stable .NET SDK. Previously the latest SDK was auto-selected regardless if it was stable or preview.
- New script names will use the next highest available number. Example: if you have "Script 1" and "Script 3" open, creating a new script will give it the name "Script 4". Before it was named "Script 2" which was confusing.
- Update OmniSharp Server to v1.39.12.
- More details are shown about errors that occur during scaffolding database connections.
- Minor styling changes:
  - The `Production` marker in the connections list is right aligned and is now just the letter `P`.
  - Fixed weird line at the top of the app logo.

## Fixes :bug: 

- Fixed Zoom menu items and their corresponding keyboard shortcuts
- Fixed an issue running scripts targeting SQLite connections on macOS
- Fixed writing `char` values to the console (#225)
- Fixed incorrect latest version label in NuGet search window (#227)
- Fixed namespace collisions when referencing some NuGet packages (#246)
- Fixed NuGet search not working when user has an unreachable feed configued in `nuget.config` (#261)

## Coming Soon :checkered_flag:

We're currently working on fulfilling requests from the issue tracker as well as a few new features:

* A new CLI tool that you can use to run C# scripts from the command line.
* Faster database queries.

## Support :heart:

If you're enjoying NetPad and would like to support open-source, please consider sponsoring the project, it really helps! Also don't forget to star the repo if you haven't already, thank you!

A special thanks to my current sponsors. Your support is _greatly_ appreciated and helps keep this project growing!

<a href="https://github.com/pelsedyr"><img src="https://github.com/pelsedyr.png" width="50px" alt="pelsedyr" /></a>&nbsp;&nbsp;<a href="https://github.com/ChristopherHaws"><img src="https://github.com/ChristopherHaws.png" width="50px" alt="ChristopherHaws" /></a>&nbsp;&nbsp;<a href="https://github.com/jdonohoo"><img src="https://github.com/jdonohoo.png" width="50px" alt="jdonohoo" /></a>&nbsp;&nbsp;<a href="https://github.com/lpreiner"><img src="https://github.com/lpreiner.png" width="50px" alt="lpreiner" /></a>&nbsp;&nbsp;<a href="https://github.com/mattjcowan"><img src="https://github.com/mattjcowan.png" width="50px" alt="mattjcowan" /></a>&nbsp;&nbsp;

**Full Changelog**: https://github.com/tareqimbasher/NetPad/compare/v0.7.2...v0.8.0

**Comment by jwallet** [comment]
vNext backend not booting (after 30150ms) on mac m2 but the default v0.8.0 does work

**Comment by flq** [comment]
For me both arm images / zip don't work on my MB air M2 - in both cases it says "is damaged and should be moved to the bin" :(

**Reply by tareqimbasher** [reply]
NetPad is not signed, so on macOS will say that when you try to run an unsigned app. See this: https://github.com/tareqimbasher/NetPad/wiki/Troubleshooting#netpad-is-damaged-and-cant-be-opened

**Reply by flq** [reply]
Yup that helps, thx. Bear in mind I have the same issue as @jwallet - the vnext version does not work for me. Same processor, Apple silicon
